src/validation.cpp

```diff
@@ -4833,18 +4833,31 @@ void ChainstateManager::CheckBlockIndex()
         return;
     }
 
-    // Build forward-pointing map of the entire block tree.
+    // Build forward-pointing data structure for the entire block tree.
+    // For performance reasons, indexes of the best header chain are stored in a vector.
+    // All remaining blocks are stored in a multimap.
+    CBlockIndex* index_walk{m_best_header};
+    const int best_hdr_height{m_best_header->nHeight};
+    std::vector<CBlockIndex*> best_hdr_chain(best_hdr_height + 1);
+    while (index_walk) {
+        best_hdr_chain[index_walk->nHeight] = index_walk;
+        index_walk = index_walk->pprev;
+    }
```

Comment:
In commit "validation: improve performance of CheckBlockIndex" (0f7963e0b306257ae5d547ba979712156c9d3a30)

This seems like this is reimplementing what's already implemented in the CChain class, and all this initialization could be replaced with:

```c++
CChain best_headers;
if (m_best_header) best_headers.SetTip(m_best_header});
```

Also code below could be simplified.

Reply:
nice! Changed it to use `CChain`.